InMobi/src/main/java/com/mopub/mobileads/InMobiAdapterConfiguration.kt

```diff
@@ -0,0 +1,243 @@
+                                || personalInfoManager.personalInfoConsentStatus == ConsentStatus.DNT) {
+                            consentObject.put(InMobiSdk.IM_GDPR_CONSENT_AVAILABLE, false)
+                        } else {
+                            consentObject.put(InMobiSdk.IM_GDPR_CONSENT_AVAILABLE, true)
+                        }
+                    } else if (canCollectPersonalInfo) {
+                        consentObject.put(InMobiSdk.IM_GDPR_CONSENT_AVAILABLE, true)
+                    } else {
+                        consentObject.put(InMobiSdk.IM_GDPR_CONSENT_AVAILABLE, false)
+                    }
+                } ?: run {
+                    consentObject.put(CONSENT_KEY, "0")
+                    consentObject.put(InMobiSdk.IM_GDPR_CONSENT_AVAILABLE, false)
+                }
+            } catch (e: JSONException) {
+                MoPubLog.log(AdapterLogEvent.CUSTOM_WITH_THROWABLE, "InMobi cannot obtain consent because of error: ", e)
+            }
+
+            return consentObject
+        }
+
+        fun initialiseInMobi(configuration: Map<String, String>, context: Context, initCompletionListener: InitCompletionListener) {
+            try {
+                val accountId = getAccountId(configuration)
+                val consentObject = InMobiGDPR.gdprConsentDictionary ?: run {
+                    getInMobiConsentDictionary()
+                }
+                InMobiSdk.init(context, accountId, consentObject) {
+                    if (it == null) {
+                        isInMobiSdkInitialised = true
+                        initCompletionListener.onSuccess()
+                    } else {
+                        initCompletionListener.onFailure(it, null)
+                    }
+                }
+            } catch (e: Exception) {
+                initCompletionListener.onFailure(null, e)
+            }
+        }
+
+        // TODO: Revision needed
+        fun getMoPubErrorCode(statusCode: InMobiAdRequestStatus.StatusCode?): MoPubErrorCode {
+            return when (statusCode) {
+                InMobiAdRequestStatus.StatusCode.INTERNAL_ERROR -> MoPubErrorCode.INTERNAL_ERROR
+                InMobiAdRequestStatus.StatusCode.REQUEST_INVALID -> MoPubErrorCode.NETWORK_INVALID_STATE
+                InMobiAdRequestStatus.StatusCode.NETWORK_UNREACHABLE -> MoPubErrorCode.NO_CONNECTION
+                InMobiAdRequestStatus.StatusCode.NO_FILL -> MoPubErrorCode.NO_FILL
+                InMobiAdRequestStatus.StatusCode.REQUEST_TIMED_OUT -> MoPubErrorCode.NETWORK_TIMEOUT
+                InMobiAdRequestStatus.StatusCode.SERVER_ERROR -> MoPubErrorCode.NETWORK_INVALID_STATE
+                else -> MoPubErrorCode.UNSPECIFIED
+            }
+        }
+
+        fun getAccountId(dict: Map<String, String>): String {
+            val accountIdErrorMessage = "Please make sure you provide correct Account ID information on MoPub UI."
```

Comment:
You can directly use this while throwing an exception. I don't see why the `val` is created.

Reply:
Let it be, they are following this kind of pattern. Assigning error messages to variables and then using them. Can raise a separate PR later but don't want to block this PR if they have done this intentionally.